Classes/Emogrifier.php

```diff
@@ -308,7 +310,9 @@ public function emogrifyBodyContent()
         }
 
         $xmlDocument = $this->createXmlDocument();
+        set_error_handler([$this, 'handleXpathError'], E_WARNING);
         $this->process($xmlDocument);
+        restore_error_handler();
```

Comment:
We have exactly the same code here and in emogrify (starting from the check for empty $this->html). Please let's move this to a separate private method to avoid the duplication.